coralproject/talk
# User Bio and Settings Tab

## What does this PR do?

This PR adds

- User Bio and Settings Tab
- Tooltip Component (for the user bio hovering the Author name)
- RestrictedContent
- Adds spinner to embedStream
- Banned/Suspended User UI (with RestrictedContent) 
- authUser !=== userData (settings, bio)

## RestrictedContent

You can wrap all the secured content within the `RestrictedContent` Container it will show a default message `This content is not available` if restricted is true. If no component is provided you can send a message.

``` javascript
// Params:
restricted: <Boolean>
restrictedComp: <ReactElement>
message: <String>
``` 

### Usage

``` javascript
<RestrictedContent restricted={!loggedIn} restrictedComp={NotLoggedIn}>
 // Restricted Stuff
</RestrictedContent>
```


## How do I test this PR?

LoggedIn
- Check the settings tab
- Add a user bio
- Hover names in the embedStream


Logged Out
 - Check the settings tab
